File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        if x < 10:
            return True
        numlen = math.ceil(log10(x+1))
        for i in range(numlen//2):
            if (x//pow(10,i))%10 != (x//pow(10,numlen-i-1))%10:
                return False
        return True
        

File: 0009-palindrome-number/0009-palindrome-number.c
bool isPalindrome(int x) {
    if (x < 0) return false;
    if (x < 10) return true;
    int numlen;
    if (x < 2147483647) {
        numlen = (int)(ceil(log10(x+1)));
    }
    else {
        numlen = (int)(ceil(log10(x)));
    }
    char str[numlen + 1];
    sprintf(str, "%d", x);
    for (int i=0;i<(numlen/2);i++){
        if (str[i] != str[numlen-i-1]) return false;
    }
    return true;
}